src/hci/gnomex/controller/GetRequestDownloadList.java

```diff
@@ -717,19 +723,22 @@ private void recurseAddFiles(Element fdNode, File f1, String requestNumber, Stri
 
 		for(int i = 0; i < files.length; i++){
 			File f = new File(fullPath + files[i]);
-			FileDescriptor fd = new FileDescriptor(requestNumber, f.getName(), f, f.getName());
+			String fName = this.getPathForZipFileName(f, requestNumber);
+			FileDescriptor fd = new FileDescriptor(requestNumber, f.getName(), f, fName);
 			fd.setDirectoryName(directoryName);
 			fd.excludeMethodFromXML("getChildren");
 			Element fileNode = fd.toXMLDocument(null, fd.DATE_OUTPUT_ALTIO).getRootElement();
 			fileNode.setAttribute("isSelected", "N");
 			fileNode.setAttribute("state", "unchecked");
 			fileNode.setAttribute("canDelete", "Y");
 			fileNode.setAttribute("canRename", "Y");
-			//fdNode.setAttribute("linkedSampleNumber", getLinkedSampleNumber(sess, fileName.substring(fileName.indexOf(Request.getBaseRequestNumber(requestNumber)))));
+			String fname = this.getPathForZipFileName(f, requestNumber);

```

Comment:
You don't need this second variable. Perhaps you could also move the "fName" and "fd" lines down here so everything is together. Maybe rename "fName" to "fPath" as well.